### ISSUE TITLE
Move away from black/white list terminology

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-filter-prune#28]
This plugin removes fields based on two criteria:
a) a list of regexes used to skip fields and values (settings currently prefixed with "whitelist")
b) a list of regexes used to indicate which fields and values to prune (settings currently prefixed with "blacklist")

The proposal where is to move away from this terminology and adopt:
* allowlist or passlist instead of "whitelist"
* denylist or blocklist instead of "blacklist"

This will be a breaking change, but that's ok.

**Comment by magnusbaeck** [comment]
@jsvd, what kind of migration path do you have in mind? Step the minor version and introduces replacement options that default to the value of the current options (which are marked as deprecated), then step the major version and get rid of the old options?

**Comment by jsvd** [comment]
exactly, @magnusbaeck! the current minor version of this plugin will be included in all 7.x versions of logstash, and the new major version will be included in logstash 8.0

**Comment by magnusbaeck** [comment]
Started working on this one. If you don't mind I'll go with `include_names`, `exclude_names`, `include_values`, and `exclude_values`. The current option names work because "whitelist" and "blacklist" are verbs, but I don't think "allowlist", "denylist", "passlist", or "blocklist" can be used as anything but a noun, which I guess is why `passlist_names` doesn't sit well.

Are there any guidelines on how to deal with option deprecation, like when to use `:deprecated => true` in the config option definition? (I found logstash-mixin-deprecation_logger_support so I'll use that for logging use of the old options.)

Same thing with replacing options, is there an elegant way of configuring a new option to default to the old option's value, i.e. basically this:

    config :new_option, ..., default => @old_option

**Comment by magnusbaeck** [comment]
@jsvd, could you please have a look at the question above?

**Comment by mjingle** [comment]
Hi all, just wanted to express my interest in this stale issue from 2020. My org would like to amend any white/blacklist mentions from our docs and this task cannot be completed until logstash reflects an updated terminology.

**Comment by jsvd** [comment]
> If you don't mind I'll go with include_names, exclude_names, include_values, and exclude_values

@magnusbaeck sorry for the very long delay... your proposal absolutely works, there's no need enforce a new replacement name when others provide a better explanation.

**Comment by mjingle** [comment]
@jsvd I haven't seen much activity from @magnusbaeck in the past month or two, so there may be a delay in resolving this issue. Would it be possible for this work to be picked up by someone else?

**Comment by magnusbaeck** [comment]
I won't have time for this for at least a couple of weeks, but I could probably dig up the half-finished work if that would help someone else. I don't remember how far I got so it might not be a huge help.
